.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -0,0 +1,228 @@
+package environment
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/gravitational/teleport/.github/workflows/ci"
+	"github.com/gravitational/trace"
+
+	"github.com/google/go-github/v37/github"
+)
+
+// Config is used to configure Environment
+type Config struct {
+	Client        *github.Client
+	Reviewers     string
+	EventPath     string
+	Token         string
+	unmarshalRevs unmarshalReviewersFn
+}
+
+// Environment contains information about the environment
+type Environment struct {
+	Client           *github.Client
+	PullRequest      *PullRequestMetadata
+	token            string
+	reviewers        map[string][]string
+	defaultReviewers []string
+	action           string
+}
+
+type unmarshalReviewersFn func(str string, client *github.Client) (map[string][]string, error)
+
+// CheckAndSetDefaults verifies configuration and sets defaults
+func (c *Config) CheckAndSetDefaults() error {
+	if c.Client == nil {
+		return trace.BadParameter("missing parameter Client.")
+	}
+	if c.Reviewers == "" {
+		return trace.BadParameter("missing parameter Reviewers.")
+	}
+	if c.EventPath == "" {
+		return trace.BadParameter("missing parameter EventPath.")
+	}
+	if c.Token == "" {
+		return trace.BadParameter("missing parameter token.")
+	}
+	if c.unmarshalRevs == nil {
+		c.unmarshalRevs = unmarshalReviewers
+	}
+	return nil
+}
+
+// New creates a new instance of Environment.
+func New(c Config) (*Environment, error) {
+	var env Environment
+	err := c.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	env.Client = c.Client
+	revs, err := c.unmarshalRevs(c.Reviewers, c.Client)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	env.reviewers = revs
+	env.defaultReviewers = revs[""]
+	err = env.SetPullRequest(c.EventPath)
```

Comment:
Instead of having a function that modifies internal env state (side-effects is an anti-pattern), I would just make a `getPullRequest` method that returns a pull request object and then set it in the env.

Also, as I mentioned somewhere above, filling out struct fields one by one (`env.Client = ...`, `env.reviewers = ..`, etc.) is not very Go idiomatic, I would just do

```go
return &Environment{
  Client: ...,
  reviewers: ...,
  ...
}, nil
```